### DISCUSSION TITLE
.net 5 backward compatiblity with .net framework?

### DISCUSSION THREAD

**Discussion by zmadkour** [dotnet/runtime#40782] [General] [answered]
This is not an issue but a want an answer if you please.

Our product has multiple projects/package involved, they are a mix between .Net framework, .net standard and .net core.

If we used a .net framework library on .net 5 application, will it run? we are having a parser based on lex-yacc .net framework package.

In other words,  Is .net 5 backward compatible with .net framework?

**Comment by danmoseley** [comment]
@ZiadMadkour a huge number of .NET Framework API also exist in .NET Core/.NET 5.0, so often libraries built for .NET Framework migrate with few or even no changes. There are some technologies that will remain .NET Framework specific, such as AppDomains, Remoting, etc, WCF Server and WWF. The latter have open source projects that may be helpful: https://github.com/CoreWCF/CoreWCF and https://github.com/UiPath-Open/corewf

A good place to start reading is https://docs.microsoft.com/en-us/dotnet/core/porting/  and it also explains the portability analyzer tool.

If you expect to share code between .NET Core/.NET 5.0 and .NET Framework, then you can either multi-target or target .NET Standard 2.0. Many libraries can be successfully converted to build against .NET Standard 2.0 without difficulty, which gives you an assembly that will work well on either.

**Comment by davidfowl** [comment, marked as answer]
This question was answered here https://github.com/dotnet/runtime/discussions/40304

**Reply by zmadkour** [reply]
I think, this question is the reverse of what I ask about :D the question asks about .net 5 library in .net framework application, my question is the other way around (i.e. a .net framework library runs in .net 5 application)

**Reply by danmoseley** [reply]
Ah I see. We did do some work to add "facades" such that many old libraries that were compiled against .NET Framework can potentially work on .NET Core. We did this mainly as a bridge to help folks get started with .NET Core until the common libraries had a chance to recompile for it. It does not work perfectly, and of course, if you use any .NET Framework API that isn't available on .NET Core, it will fail at runtime - the compiler didn't protect you.

If you want to reuse code in both .NET Framework and .NET Core, your best bet is to create libraries targeting .NET Standard 2.0. If that isn't enough, you can cross target.
This explains it all
https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting

**Reply by zmadkour** [reply]
That's exactly what I needed, thank you

**Comment by zmadkour** [comment]
Thank you All :D got it.

**Reply by danmoseley** [reply]
Please mark answered 😀
